crawl-ref/source/dungeon.cc

```diff
@@ -3777,12 +3776,20 @@ static void _builder_monsters()
     for (int i = 0; i < mon_wanted; i++)
     {
         mgen_data mg;
-        if (!in_pan)
+        // Chance to generate the monster awake, but away from level stairs.
+        if (player_in_connected_branch()
+            // Exclude D:1 from generating these awake monsters.
+            && (!player_in_branch(BRANCH_DUNGEON) || you.depth > 1)
```

Comment:
Can you use absdepth here to avoid needing to check both branch and level? The conditional is a little bit hard to read, and I think nothing has the same absdepth as D:1.

Reply:
Yeah, I can check `env.absdepth0`, so I'll do that